app/com/linkedin/drelephant/util/Utils.java

```diff
@@ -234,6 +237,79 @@ public static String truncateField(String field, int limit, String appId) {
   }
 
   /**
+   * Convert a millisecond duration to a string format
+   *
+   * @param millis A duration to convert to a string form
+   * @return A string of the form "X:Y:Z Hours".
+   */
+  public static String getDurationBreakdown(long millis) {
+
+    long hours = TimeUnit.MILLISECONDS.toHours(millis);
+    millis -= TimeUnit.HOURS.toMillis(hours);
+    long minutes = TimeUnit.MILLISECONDS.toMinutes(millis);
+    millis -= TimeUnit.MINUTES.toMillis(minutes);
+    long seconds = TimeUnit.MILLISECONDS.toSeconds(millis);
+
+    StringBuilder sb = new StringBuilder(64);
+
+    sb.append(hours);
+    sb.append(":");
+
+    sb.append(minutes);
+    sb.append(":");
+
+    sb.append(seconds);
+    sb.append(" Hours");
+
+    return (sb.toString());
```

Comment:
Can we have the time returned as 00:01:44 Hours rather than 0:1:44 Hours?

String.format("%02d:%02d:%02d:%d", hours, minutes, seconds);